docs/source/integrations/working_with_sql.md

```diff
@@ -0,0 +1,207 @@
+```
+
+- **Mechanism**
+  Spark’s DataFrame API over JDBC. Leverages predicate pushdown so filters/projections occur in-database.
+
+- **Pros**
+
+  - Distributed, lazy compute
+  - Leverages cluster power
+- **Cons**
+
+  - JVM + Spark cluster required
+  - Overkill for smaller workloads
+
+---
+
+### 3. Ibis (modern)
+
+```{tip}
+Please also check out our [blog on building scalable data pipelines with Kedro and Ibis](https://kedro.org/blog/building-scalable-data-pipelines-with-kedro-and-ibis), [SQL data processing in Kedro ML pipelines](https://kedro.org/blog/sql-data-processing-in-kedro-ml-pipelines), and our [PyData London talk](https://www.youtube.com/watch?v=ffDHdtz_vKc).
+```
+
+```yaml
+ibis_orders:
+  type: ibis.TableDataset
+  backend: snowflake
+  credentials: snowflake_creds
+  table: orders
+  save_args:
+    materialized: table
+    database: my_db.DEV
+```
+
+- **Mechanism**
+  Python expression trees → compiled SQL at runtime.
+- **Key highlights**
+
+  - [SQLGlot](https://github.com/tobymao/sqlglot) translation across 20+ engines
+  - Support for extensions (DuckDB geospatial, PostGIS, Snowflake UDFs and more)
+  - Full lazy push-down (filters, joins, aggregates)
+  - Pluggable materialization (table/view/CTE)
+  - No JVM/Spark overhead
+  - In-memory/mock backends for testing
+- **Pros**
+
+  - Consistent API everywhere
+  - Retains engine optimizations
+  - Excellent testability & reproducibility
+- **Cons**
+
+  - Some backends may miss niche SQL operations
+  - Migration effort for legacy pipelines may not justify moving functional code, prioritise new pipelines / projects
+
+```{warning}
+Consult the [Ibis support matrix](https://ibis-project.org/backends/support/matrix) to verify that needed SQL functions are available. You can still write raw SQL for missing features.
+```
+
+### Recommended Ibis development workflow
+
+<details>
+<summary>Iterate locally for instant, low cost feedback. Deploy to production your warehouse for scale—ensuring consistency, testability, and zero drift between environments.</summary>
+
+#### DuckDB (or SQLlite) for Development & [CI](https://en.wikipedia.org/wiki/Continuous_integration)
+
+- In-memory, zero-config engine
+- Fast startup for local iteration and unit tests
+   ```yaml
+   # conf/base/catalog.yml
+   orders:
+     type: ibis.TableDataset
+     backend: duckdb
+     filepath: data/03_primary/dev.duckdb
+     table: orders
+   ```
+
+#### Warehouse Backend for Production & [CD](https://en.wikipedia.org/wiki/Continuous_delivery) (for example Snowflake or BigQuery)**
+
+- Simply swap your catalog’s backend and credentials
+- Same Ibis expressions compile to your warehouse’s SQL
+- Leverage scale, governance, cost controls
+   ```yaml
+   # conf/prod/catalog.yml
+   orders:
+     type: ibis.TableDataset
+     backend: snowflake         # or bigquery and more.
+     credentials: warehouse_creds
+     table: analytics.orders
+   ```
+
+Example pattern:
+
+```mermaid
```

Comment:
```
generating indices... /home/docs/checkouts/readthedocs.org/user_builds/kedro/checkouts/4706/docs/source/integrations/working_with_sql.md:169: WARNING: Pygments lexer name 'mermaid' is not known 
```

Unfortunately, our mermaid situation is a bit complex https://github.com/kedro-org/kedro/wiki/Render-Mermaid-diagrams

Reply:
oh we just need to generate a PNG? I'll wait for the review and do this